teuthology/orchestra/remote.py

```diff
@@ -69,10 +71,13 @@ def mktemp(self, suffix=None, parentdir=None, data=None):
         Returns: the path of the temp file created.
         """
         args = ['mktemp']
+
         if suffix:
             args.append('--suffix=%s' % suffix)
-        if parentdir:
-            args.append('--tmpdir=%s' % parentdir)
+        if parentdir is None:
+            parentdir = misc.get_tmpdir()
+            self.sh(f"mkdir -p {parentdir}")
+        args.append('--tmpdir=%s' % parentdir)
```

Comment:
IMHO, it would be simpler to modify `mktemp()` to set `parentdir`'s default value to `config.get(test_tmpdir)` and add an assert check for `parentdir` that `/tmp/` is not the parent dir for the temp files.

Reply:
it is simpler, but now, when I looked into the details of the cause of this PR even appeared I think it is just incorrect... please stop breaking framework with workarounds instead of fixing wrong algorithms in the tests, if kernel has the feature not allowing to create files in directory own by someone else even for root, then just stop doing this, and do make another bottlenecks